src/components/Asset/AssetActions/Trade/FormTrade.tsx

```diff
@@ -64,32 +73,70 @@ export default function FormTrade({
     .defined()
 
   async function handleTrade(values: FormTradeData) {
+    if (!web3 || !asset || !poolInfo || !values) return
+
     try {
+      const poolInstance = new Pool(web3)
+
+      const swapMarketFeeIn = await poolInstance.getMarketFees(
+        asset.accessDetails.addressOrId,
+        poolInfo.baseTokenAddress
+      )
+
+      const swapMarketFeeOut = await poolInstance.getMarketFees(
+        asset.accessDetails.addressOrId,
+        poolInfo.datatokenAddress
+      )
+
+      const tokenInOutMarket: TokenInOutMarket = {
+        tokenIn: poolInfo.baseTokenAddress,
+        tokenOut: poolInfo.datatokenAddress,
+        marketFeeAddress: appConfig.consumeMarketPoolSwapFee
+      }
+      const tokenOutMarket: TokenInOutMarket = {
+        tokenIn: poolInfo.datatokenAddress,
+        tokenOut: poolInfo.baseTokenAddress,
+        marketFeeAddress: appConfig.consumeMarketPoolSwapFee
+      }
+
       const impact = new Decimal(
         new Decimal(100).sub(new Decimal(values.slippage))
       ).div(100)
       const precision = 15
-      // const tx =
-      //   values.type === 'buy'
-      //     ? await ocean.pool.buyDTWithExactOcean(
-      //         accountId,
-      //         price.address,
-      //         new Decimal(values.datatoken)
-      //           .mul(impact)
-      //           .toFixed(precision)
-      //           .toString(),
-      //         new Decimal(values.ocean).toFixed(precision).toString()
-      //       )
-      //     : await ocean.pool.sellDT(
-      //         accountId,
-      //         price.address,
-      //         new Decimal(values.datatoken).toFixed(precision).toString(),
-      //         new Decimal(values.ocean)
-      //           .mul(impact)
-      //           .toFixed(precision)
-      //           .toString()
-      //       )
-      // setTxId(tx?.transactionHash)
+
+      const amountsInOutMaxFee: AmountsInMaxFee = {
+        tokenAmountIn: new Decimal(values.datatoken)
+          .mul(impact)
+          .toFixed(precision)
+          .toString(),
+        minAmountOut: '2',
```

Comment:
I guess we don't know yet how to calculate this hardcoded number?

Reply:
yes, not sure if i missed something but i think it's somehow related to [this issue](https://github.com/oceanprotocol/market/issues/1027)

Reply:
commented bellow, no actions needed yet